### ISSUE TITLE
undefined: time.UnixMilli when updating

### ISSUE THREAD

**Issue by zogith** [cointop-sh/cointop#260]
Xubuntu 20.04 with **golang-go:amd64/focal 2:1.13~1ubuntu2 uptodate** installed

updating the installed compiled binary (or trying to install fresh) fails with this error, on 2 different PCs for me:

```
go get -u github.com/cointop-sh/cointop
# github.com/cointop-sh/cointop/pkg/timedata
go/src/github.com/cointop-sh/cointop/pkg/timedata/timedata.go:43:14: undefined: time.UnixMilli
go/src/github.com/cointop-sh/cointop/pkg/timedata/timedata.go:44:14: undefined: time.UnixMilli
go/src/github.com/cointop-sh/cointop/pkg/timedata/timedata.go:52:13: undefined: time.UnixMilli
go/src/github.com/cointop-sh/cointop/pkg/timedata/timedata.go:53:13: undefined: time.UnixMilli
go/src/github.com/cointop-sh/cointop/pkg/timedata/timedata.go:68:16: undefined: time.UnixMilli
```

**Comment by lyricnz** [comment]
Looks like this was added in golang 1.17 https://pkg.go.dev/time#UnixMilli

Which version are you running?  (run `go version`)

**Comment by lyricnz** [comment]
@miguelmota what's cointop minimum golang version?  Would be simple enough to use a utility function for this if <1.17

```
func UnixMilli(msec int64) Time {
	return Unix(msec/1e3, (msec%1e3)*1e6)
}
```

**Comment by zogith** [comment]
> Looks like this was added in golang 1.17 https://pkg.go.dev/time#UnixMilli
> 
> Which version are you running? (run `go version`)

```
go version
go version go1.13.8 linux/amd64
```

**Comment by lyricnz** [comment]
The current code needs go 1.17.  If you don't want to build it yourself, maybe you could just use the pre-build binary?

https://github.com/cointop-sh/cointop/releases/download/v1.6.9/cointop_1.6.9_linux_amd64.tar.gz

**Comment by zogith** [comment]
On Xubuntu 20.04 (and presumably any other Ubuntu variant on the current LTS release) the prebuilt binary gives:

`./cointop: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by ./cointop)`

but installing go 1.17 on the box could be an acceptable solution.

**Comment by zogith** [comment]
I have installed go 1.17 using the Ubuntu "snap" package (not my first choice but I prefer this over installing from source) using the command described in https://linuxhint.com/install-go-ubuntu-2/
and then 
```
go install  github.com/cointop-sh/cointop@v1.6.9
```
worked for me, I now have a shiny new working cointop, thanks.

**Comment by lyricnz** [comment]
#101 is about the glibc dep. I don't have such a box, or idea how to build such a binary in CI.

**Comment by zogith** [comment]
https://docs.cointop.sh/install/ mentions go 1.12 as the minimum, and references an install command syntax that will not work with  go 1.17

**Comment by lyricnz** [comment]
patches welcome!

**Comment by lyricnz** [comment]
It wouldn't be too hard to use a utility version, rather than the golang 1.17 version, of UnixMilli. But only worthwhile if the project wants to support older golang, which is not my call.

**Comment by miguelmota** [comment]
@lyricnz I'm in favor of using the latest stable version of Go. I would recommend installing cointop from a package manager or using a pre-compiled binary that is available for systems where Go is not up-to-date or available.

**Comment by lyricnz** [comment]
Merged PR #301
